saleor/graphql/attribute/utils/type_handlers.py

```diff
@@ -423,14 +423,19 @@ def pre_save_value(self, instance: T_INSTANCE) -> list[tuple]:
 
 
 class ReferenceAttributeHandler(AttributeTypeHandler):
-    """Handler for Reference attribute type."""
+    """Handler for Reference and Single Reference attribute type."""
+
+    def get_references(self) -> Sequence[str | None]:
```

Comment:
Does typing correct? It looks like you always return `Sequence[str]`

Reply:
Or maybe, we expect to have `self.values_input.references == [None, None, None]` (if this is true, then I think we need to make strip somewhere to drop nulls)

Reply:
Will update in another PR